File: Data Structures & Algorithms/find-peak-element/submission-1.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int l = 0;
        int r = nums.size()-1;
        int mid;
        int peak = 0;
        int n = nums.size();
        while(l < r){
            mid = l+(r-l)/2;
            if(nums[mid] > nums[mid+1]){
                r = mid;
            }
            else
            {
                l = mid+1;
            }
        
        }
        return l;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        
        n = len(nums)
        subsets = []
        nums.sort()
        def backTrack(i,subset):
            if i==n:
                subsets.append(subset.copy())
                return

            subset.append(nums[i])
            backTrack(i+1,subset)
            subset.pop()
            #skipping all the duplicates: 
            while((i+1)<n and nums[i]==nums[i+1]):
                i+=1 
            backTrack(i+1,subset)
        
        backTrack(0,[])
        return subsets
            